helpers/utils.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+from typing import List, Dict
+import re
+import pwd
+import subprocess
+import logging
+import io
+
+USER_HOME_DIR_ENT_IDX = 5
+EXIT_CODE_BAD_KUBECONFIG = -100
+EXIT_CODE_BAD_CONF_ARG = -200
+
+def generate_spark_default_conf() -> Dict:
+    generated_defaults = dict()
+    generated_defaults['spark.kubernetes.container.image'] = 'docker.io/averma32/sparkpy6:latest'
+    generated_defaults['spark.kubernetes.container.image.pullPolicy'] = 'IfNotPresent'
+    USER_HOME_DIR = pwd.getpwuid(os.getuid())[USER_HOME_DIR_ENT_IDX]
+    SCALA_HIST_FILE_DIR = os.environ.get('SNAP_USER_COMMON', USER_HOME_DIR)
+    generated_defaults['spark.driver.extraJavaOptions'] = f'-Dscala.shell.histfile={SCALA_HIST_FILE_DIR}/.scala_history'
+    return generated_defaults
+
+def read_property_file(name: str) -> Dict :
+    defaults = dict()
+    with open(name) as f:
+        for line in f:
+            kv = list(filter(None, re.split('=| ', line.strip())))
+            k = kv[0]
+            v = '='.join(kv[1:])
+            defaults[k] = os.path.expandvars(v)
+    return defaults
+
+def write_property_file(fp: io.TextIOWrapper, props: Dict) -> None:
+    for k in props.keys():
+        v = props[k]
+        fp.write(f"{k}={v.strip()}\n")
+
+def merge_configurations(dictionaries_to_merge: List[Dict]) -> Dict:
+    result = dict()
+    for override in dictionaries_to_merge:
+        result.update(override)
+    return result
+
+def get_snap_defaults_conf_file() -> str:
+    SPARK_HOME = os.environ.get('SPARK_HOME', os.environ.get('SNAP'))
+    SPARK_CONF_DIR = os.environ.get('SPARK_CONF_DIR', f"{SPARK_HOME}/conf")
+    SPARK_CONF_DEFAULTS_FILE = os.environ.get('SNAP_SPARK_ENV_CONF', f"{SPARK_CONF_DIR}/spark-defaults.conf")
+    return SPARK_CONF_DEFAULTS_FILE
+
+def get_static_defaults_conf_file() -> str:
+    return os.environ.get('SNAP') + '/conf/spark-defaults.conf'
+
+def get_dynamic_defaults_conf_file() -> str:
```

Comment:
I would call this "get_env_defaults_conf_file". Dynamic is somewhat confusing since the dynamic part will be the task DPE-836, and I would just to stick to the naming we had in the slides